### PR TITLE
Add selectors to config

### PR DESCRIPTION
**Pull request by davidrapson**
This PR adds the selectors `[data-gumshoe] a` and `[data-gumshoe-header]` to the settings object as it's useful to be able to override these.

Thanks,
David

---

As an aside I had a fair amount of trouble with the build as a lot of packages fail to install under node `0.12.x`. I had to upgrade the packages to get it to work and disable a few unneeded(?) gulp tasks. Happy to raise another PR to update the build dependencies if that would help.

### CODE REVIEW

File: dist/js/gumshoe.js [hunk 1]
```diff
@@ -24,13 +24,15 @@
 	//
 
 	var gumshoe = {}; // Object for public APIs
-	var supports = !!document.querySelector && !!root.addEventListener; // Feature test
+	var supports = !!document.querySelectorAll && !!root.addEventListener; // Feature test
```
Comment (davidrapson) [inline dist/js/gumshoe.js]: I updated this to check for `querySelectorAll` as your main `navLinks` selector requires it so it felt better to update this check to match.

```js
var navLinks = document.querySelectorAll( settings.selector );
```

**Comment by cferdinandi** [comment]
1. Thanks for pointing out the build issues. I've been running the same build for so long, I hadn't even noticed. I'll update accordingly.
2. Any compelling reason to add this functionality? What's the use case here?
3. `querySelector` support is identical to `querySelectorAll` support, so I'm not sure this is a necessary change.

**Comment by davidrapson** [comment]
> Any compelling reason to add this functionality? What's the use case here?

It's useful to be able to control what is used as the selector, gives a little more control over what elements are included (e.g., not all `<a>` elements in a container). Also has the benefit of being able to use it in cases where you don't control the markup for some reason (not my use case here, but still useful).

> querySelector support is identical to querySelectorAll support, so I'm not sure this is a necessary change.

Doh! You are completely right, not sure where I'd got that from, will revert that change.

**Comment by davidrapson** [comment]
@cferdinandi Flattened down the PR to the bare minimum, only adding the selectors to the config. Are you happy with the value of the change as it is now?

**Comment by cferdinandi** [comment]
@davidrapson Dude, this is perfect. Thanks! I'm going to update the version number, push a git Tag, and fix a small typo in the readme. but THANK YOU!

Most people don't both running the build or updating the readme, so major kudos for doing that!

**Comment by davidrapson** [comment]
No problem at all. Hah, yeah `headerSelecto` should probably have an `r` shouldn't it.

**Comment by cferdinandi** [comment]
All good, man. Thanks for sticking it out. Appreciate the PR!

**Pull request merged**